src/Components/controlToggle.js

```diff
@@ -0,0 +1,38 @@
+import controls from "../services/controls.service";
+
+export const controlToggle = (state, key, checked) => {
+  let result = [key];
+  const pressedControl = controls.find(control => control.key === key);
+  const masterKey = pressedControl.masterKey;
+
+  if (masterKey && checked) {
+    // A sub checkbox enabled. Enable master.
+    result.push(masterKey);
+  }
+
+  if (masterKey && !checked) {
+    const subBoxes = controls.filter(obj => obj.masterKey === masterKey);
+    const subStates = subBoxes.map(sub => state[sub.key]);
+    const offNumber = subStates.reduce(
+      (acc, isOn) => (isOn ? acc : acc + 1),
+      1
+    );
+
+    if (offNumber === subBoxes.length) {
+      // All sub checkboxes off. Disable master.
```

Comment:
try something like this:
```suggestion
  const allOff = subBoxes.every(sub => !state[sub.key])
  
  if( allOff) {
    // All sub checkboxes off. Disable master.
  }
```

Reply:
The state have not been updated at this point so the pressed checkbox will still be on in the state map. The accumulator starts on 1 to counter this.

Reply:
eh ok